sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala

```diff
@@ -45,6 +44,14 @@ class ClosableByteArrayInputStream(buf: Array[Byte]) extends ByteArrayInputStrea
 
 class UnsafeRowSerializerSuite extends SparkFunSuite with LocalSparkContext {
 
+  override def beforeAll() {
+    super.beforeAll()
+    // This test suite calls `UnsafeProjection.create` which accesses `SQLConf.get`, we should make
+    // sure active session is cleaned so that `SQLConf.get` won't refer to a stopped session.
+    SparkSession.clearActiveSession()
+    SparkSession.clearDefaultSession()
```

Comment:
should we actually do this in `LocalSparkContext.resetSparkContext()` (which is called in `afterEach`)?

I worry there are a lot more places we should be doing this, eg. `MLlibTestSparkContext.afterAll()` only calls `SparkSession.clearActiveSession()`.

this class does seem to be one of the major sources of flakiness so I'm also OK with just getting this in for now.

Reply:
the problem is that, `LocalSparkContext` is in core module not SQL, so we can't access `SparkSession` there.

Maybe we should use `LocalSparkSession` for this test suite.

Reply:
@squito @cloud-fan : we are facing similar issues in our own Spark tests. Are you saying just closing the sparkSession is not enough ?

Reply:
@praetp yeah I think so, you probably also need to call `clearActiveSession()` and `clearDefaultSession()` in your `afterEach()` in your tests (or similar).

If that doesn't work, might be best to open a thread on dev@spark.apache.org with more details.

Reply:
Seems like a major resource leakage to me then..